hcl2template/types.packer_config.go

```diff
@@ -18,7 +18,7 @@ type PackerConfig struct {
 	Basedir string
 
 	// Available Source blocks
-	Sources map[SourceRef]*SourceBlock
+	Sources map[SourceRef]SourceBlock
```

Comment:
it's not obvious to me why you switched away from pointers here.

Reply:
To avoid changing the 'original source' when we set one specifically in a build; because it is pointer references

Reply:
Another option would have been to duplicate the referenced object but this is prone to errors and would have required too much attention from me; and since I originally made them pointers as a convenience way to do something like this:

```go
if existing := cfg.Sources[ref]; existing != nil {
```
I thought it is more convenient/simple here to change that if so that the overall code remains simple.

Reply:
Makes sense, thanks for the clarification.